### PR TITLE
C3 nohahesham f1

### PR DESCRIPTION
**Pull request by nohahesham**
#95
#78

C3.F1 and C3.S20
@Fattouh92 
@SalmaFG 
@m0hamed 
@NourhanMohamed

### CODE REVIEW

File: arability/app/models/my_subscription.rb [hunk 1]
```diff
@@ -1,4 +1,69 @@
...
+    return true
+    else
+      return false
+    end
+  end
+
+#   def self.decrement!(attribute, by = 1,dev_id)
+#     mySubscription = MySubscription.joins(:developer).where(:developer_id => dev_id).first
+#   decrement(attribute, by).update_attribute(attribute, self[attribute])
+# end
```
Comment (m0hamed) [inline arability/app/models/my_subscription.rb]: please delete commented out lines

File: arability/app/models/project.rb [hunk 2]
```diff
@@ -1,4 +1,3 @@
 class Project < ActiveRecord::Base
-  belongs_to :developer
```
Comment (m0hamed) [inline arability/app/models/project.rb:2]: why did you remove this line

File: arability/app/models/subscription_model.rb [hunk 3]
```diff
@@ -1,3 +1,9 @@
 class SubscriptionModel < ActiveRecord::Base
+	has_many :my_subscriptions
   # attr_accessible :title, :body
+#def find_add_limit
```
Comment (m0hamed) [inline arability/app/models/subscription_model.rb]: remove commented out lines

**Comment by abulazm22** [comment]
@nohahesham noha the word_search_permission, word_follow_permission and word_add_permission should come from something. Also, return true and return false should go to something. Use <<>> instead of just <>. As for the developer in the first 3 diagrams it should have a box underneath it. If you are using just if condition use opt instead of alt and remove the else part

File: arability/app/views/developer/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+<h1>Developer#index</h1>
```
Comment (m0hamed) [inline arability/app/views/developer/index.html.erb]: if you are not using this file then delete it

File: arability/app/views/projects/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<h1>Projects</h1>
+
+<table>
+  <tr>
+    <th>Name</th>
+    <th></th>
```
Comment (m0hamed) [inline arability/app/views/projects/index.html.erb]: why an empty <th>

File: arability/config/routes.rb [hunk 6]
```diff
@@ -1,6 +1,19 @@
 Arability::Application.routes.draw do
+
+
+  resources :projects
+
+
+  resources :developer
+
+  root :to => "developer#index"
```
Comment (m0hamed) [inline arability/config/routes.rb]: why did you change the route to developer#index

File: arability/db/migrate/20130326160511_add_limits_to_subscription_model.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+class AddLimitsToSubscriptionModel < ActiveRecord::Migration
```
Comment (m0hamed) [inline arability/db/migrate/20130326160511_add_limits_to_subscription_model.rb:1]: All migrations should be on separate branch.

**Comment by NourhanMohamed** [comment]
Both stories should not be in one pull request. Each one should have a separate pull request, linked with its issue view the hash symbol followed by the issue number.

**Comment by abulazm22** [comment]
Noha add integer: before dev_id and :boolean after the method parameters

File: arability/app/views/projects/index.html.erb (hunk 5 above)
Comment (nohahesham) [inline arability/app/views/projects/index.html.erb]: i dont get it?

File: arability/app/controllers/developer_controller.rb [hunk 8]
```diff
@@ -0,0 +1,6 @@
+class DeveloperController < ApplicationController
+  def index
```
Comment (NourhanMohamed) [inline arability/app/controllers/developer_controller.rb]: Is this method left empty on purpose??!

File: arability/app/controllers/projects_controller.rb [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+  def index
+    @projects = Project.all
+
+    respond_to do |format|
+      format.html # index.html.erb
+      format.json { render json: @projects }
+    end
+  end
+
+  # GET /projects/1
```
Comment (NourhanMohamed) [inline arability/app/controllers/projects_controller.rb]: I know these comments are auto generated under scaffolding, but they should be removed and replaced with appropriate documentation. They should be totally eliminated if they have nothing to do with the story, too.

File: arability/app/models/developer.rb [hunk 10]
```diff
@@ -1,4 +1,10 @@
 class Developer < ActiveRecord::Base
-  belongs_to :gamer
+  belongs_to :gamer 
+  has_one :my_subscription
+
+
+
+  
   # attr_accessible :title, :body
```
Comment (NourhanMohamed) [inline arability/app/models/developer.rb]: Is that a comment really or someone forgot to comment it? If it's an intended comment then please delete it

File: arability/app/models/my_subscription.rb [hunk 11]
```diff
@@ -1,4 +1,69 @@
 class MySubscription < ActiveRecord::Base
-  attr_accessible :developer, :word_add, :word_follow, :word_search
+  attr_accessible :developer_id, :word_add, :word_follow, :word_search
   belongs_to :developer
+  belongs_to :subscription_model
+
+  
+  def self.decrement_limit(dev_id)
```
Comment (NourhanMohamed) [inline arability/app/models/my_subscription.rb]: use class << self instead of self. 
you can find here an illustration on how this can be done: http://www.railstips.org/blog/archives/2009/05/11/class-and-instance-methods-in-ruby/

File: arability/app/models/my_subscription.rb [hunk 12]
```diff
@@ -1,4 +1,69 @@
...
+    end
+  end
+  #Description:takes the developer id and checks wether the developer's word add limit has been 
+#reached ,if its not then it is greater than zero and permission is given to add and visual stats for words
+#by returning true else return false and permission denied.
+#params:developer id
+#success:permission is given if the developer didnt exceed the add limit
+#fail:none
+
+  def self.add_word_permission(dev_id)
```
Comment (NourhanMohamed) [inline arability/app/models/my_subscription.rb]: again use the class << self instead of self. thingy

File: arability/app/views/projects/index.html.erb (hunk 5 above)
Comment (NourhanMohamed) [inline arability/app/views/projects/index.html.erb]: why are you having an empty tag <th></th>

File: arability/app/views/projects/index.html.erb [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+    <th></th>
+    <th></th>
+  </tr>
+
+<% @projects.each do |project| %>
+  <tr>
+    <td><%= project.name %></td>
+
```
Comment (NourhanMohamed) [inline arability/app/views/projects/index.html.erb]: avoid excessive un-necessary blank lines

**Comment by abulazm22** [comment]
@nohahesham please add story number beside title and the fourth UML doesn't have a title at all. please check that.
Also add activation boxes under developer and type of input and output beside every function as i told you before

File: arability/app/controllers/projects_controller.rb [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+        #format.html { render action: "edit" }
+        #format.json { render json: @project.errors, status: :unprocessable_entity }
+      #end
+    #end
+  #end
+
+  # DELETE /projects/1
+  # DELETE /projects/1.json
+
+ #Description:finds the project by its id then destroys it
```
Comment (salmafg) [inline arability/app/controllers/projects_controller.rb]: Author name not mentioned.

File: arability/app/models/my_subscription.rb [hunk 15]
```diff
@@ -1,4 +1,69 @@
 class MySubscription < ActiveRecord::Base
-  attr_accessible :developer, :word_add, :word_follow, :word_search
+  attr_accessible :developer_id, :word_add, :word_follow, :word_search
   belongs_to :developer
+  belongs_to :subscription_model
+
+
```
Comment (salmafg) [inline arability/app/models/my_subscription.rb]: No documentation.

File: arability/app/models/my_subscription.rb [hunk 16]
```diff
@@ -1,4 +1,69 @@
...
+    else
+      return false
+    end
+  end
+
+#   def self.decrement!(attribute, by = 1,dev_id)
+#     mySubscription = MySubscription.joins(:developer).where(:developer_id => dev_id).first
+#   decrement(attribute, by).update_attribute(attribute, self[attribute])
+# end
+
```
Comment (salmafg) [inline arability/app/models/my_subscription.rb]: Author not mentioned.

File: arability/app/models/my_subscription.rb [hunk 17]
```diff
@@ -1,4 +1,69 @@
...
+#Description:takes the developer id and checks wether the developer's word search limit has been
```
Comment (salmafg) [inline arability/app/models/my_subscription.rb]: Make sure U follow the format we agreed on with indentation.

File: arability/app/models/my_subscription.rb [hunk 18]
```diff
@@ -1,4 +1,69 @@
...
+
+  def self.add_word_permission(dev_id)
+     mySubscription = MySubscription.joins(:developer).where(:developer_id => dev_id).first
+  	if mysubscription.word_add > 0
+  		return true
+  	else 
+  		return false
+    end  	
+  end
+  #Description:takes the developer id and checks wether the developer's word follow limit has been
```
Comment (salmafg) [inline arability/app/models/my_subscription.rb]: Author not mentioned and format not applied.

File: arability/app/controllers/developer_controller.rb (hunk 8 above)
Comment (nohahesham) [inline arability/app/controllers/developer_controller.rb]: no i will remove

**Comment by nohahesham** [comment]
plz review 
@Fattouh92 
@SalmaFG 
@m0hamed 
@NourhanMohamed  plz check the class self

File: arability/app/models/my_subscription.rb [hunk 19]
```diff
@@ -1,4 +1,87 @@
...
+  attr_accessible :developer_id, :word_add, :word_follow, :word_search
   belongs_to :developer
+  belongs_to :subscription_model
+
+  class << self
+  def decrement_limit(dev_id)
+  	mySubscription = MySubscription.joins(:developer).where(:developer_id => dev_id).first
+    old_limit = mySubscription.word_search
+  	if  mySubscription.word_search > 0
+  	mySubscription.word_search = old_limit - 1
```
Comment (m0hamed) [inline arability/app/models/my_subscription.rb]: please indent the lines inside the if statement

File: arability/app/models/my_subscription.rb [hunk 20]
```diff
@@ -1,4 +1,87 @@
...
+#   reached ,if its not then it is greater than zero and permission is given to search for words
+#   by returning true else return false and permission denied.
+# params:
+#   developer id
+# success:
+#   permission is given if the developer didnt exceed the search limit
+# fail:
+#   none
+  class << self
+  def search_word_permission(dev_id)
```
Comment (m0hamed) [inline arability/app/models/my_subscription.rb]: please indent the entire method inside class << self

File: arability/app/models/my_subscription.rb [hunk 21]
```diff
@@ -1,4 +1,87 @@
...
+#   reached ,if its not then it is greater than zero and permission is given to add and visual stats for words
+#   by returning true else return false and permission denied.
+# params:
+#   developer id
+# success:
+#   permission is given if the developer didnt exceed the add limit
+# fail:
+#   none
+  class << self
+  def add_word_permission(dev_id)
```
Comment (m0hamed) [inline arability/app/models/my_subscription.rb]: please indent the function correctly

File: arability/app/models/my_subscription.rb [hunk 22]
```diff
@@ -1,4 +1,87 @@
...
+#   reached ,if its not then it is greater than zero and permission is given to follow words
+#   by returning true else return false and permission denied.
+# params:
+#   developer id
+# success:
+#   permission is given if the developer didnt exceed the follow word limit
+# fail:
+#   none
+  class << self
+  def follow_word_permission(dev_id)
```
Comment (m0hamed) [inline arability/app/models/my_subscription.rb]: please indent the function correctly

File: arability/app/models/subscription_model.rb [hunk 23]
```diff
@@ -1,3 +1,7 @@
 class SubscriptionModel < ActiveRecord::Base
+	has_many :my_subscriptions
   # attr_accessible :title, :body
```
Comment (m0hamed) [inline arability/app/models/subscription_model.rb]: if you are commenting out a line then delete it entirely

File: arability/app/views/projects/show.html.erb [hunk 24]
```diff
@@ -0,0 +1,10 @@
+<p id="notice"><%= notice %></p>
+
+<p>
+  <b>Name:</b>
+  <%= @project.name %>
+</p>
+
```
Comment (m0hamed) [inline arability/app/views/projects/show.html.erb]: avoid unnecessary blank lines

**Comment by m0hamed** [comment]
please tag the pull request with the issue number and correct the labels on the issue

File: arability/app/controllers/projects_controller.rb [hunk 25]
```diff
@@ -0,0 +1,71 @@
...
+        format.json { render json: @project.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  
+
+  # DELETE /projects/1
+  # DELETE /projects/1.json
+ # author:Noha
```
Comment (salmafg) [inline arability/app/controllers/projects_controller.rb]: Change author name to Noha Hesham

File: arability/app/models/my_subscription.rb [hunk 26]
```diff
@@ -1,4 +1,87 @@
...
+    else
+      return false
+    end
+  end
+  end
+
+
+
+
+#author:Noha
```
Comment (salmafg) [inline arability/app/models/my_subscription.rb]: Change author name to Noha Hesham

File: arability/app/models/my_subscription.rb [hunk 27]
```diff
@@ -1,4 +1,87 @@
...
+  def add_word_permission(dev_id)
+     mySubscription = MySubscription.joins(:developer).where(:developer_id => dev_id).first
+  	if mysubscription.word_add > 0
+  		return true
+  	else 
+  		return false
+    end  	
+  end
+  end
+# author:Noha
```
Comment (salmafg) [inline arability/app/models/my_subscription.rb]: Change author name to Noha Hesham

File: arability/app/controllers/projects_controller.rb [hunk 28]
```diff
@@ -0,0 +1,71 @@
+class ProjectsController < ApplicationController
```
Comment (NourhanMohamed) [inline arability/app/controllers/projects_controller.rb:1]: There are multiple methods here resulting from a scaffold action, remove the extra methods that don't relate to your work

File: arability/app/controllers/projects_controller.rb [hunk 29]
```diff
@@ -0,0 +1,71 @@
...
+ # Description:
+ #   finds the project by its id then destroys it
+ # params:
+ #   none
+ # success:
+ #   a pop up appears and makes sure the user wants to delete the project
+ #   by choosing ok the project is successfully deleted 
+ # failure:
+ #    project is not deleted
+
```
Comment (NourhanMohamed) [inline arability/app/controllers/projects_controller.rb]: remove the 2 extra new lines

File: arability/app/models/developer.rb [hunk 30]
```diff
@@ -1,4 +1,10 @@
 class Developer < ActiveRecord::Base
-  belongs_to :gamer
-  # attr_accessible :title, :body
+  belongs_to :gamer 
+  has_one :my_subscription
+
+
```
Comment (NourhanMohamed) [inline arability/app/models/developer.rb]: remove extra new lines

File: arability/app/models/my_subscription.rb [hunk 31]
```diff
@@ -1,4 +1,84 @@
 class MySubscription < ActiveRecord::Base
-  attr_accessible :developer, :word_add, :word_follow, :word_search
+  attr_accessible :developer_id, :word_add, :word_follow, :word_search
   belongs_to :developer
+  belongs_to :subscription_model
+
+  class << self
+   def decrement_limit(dev_id)
+  	mySubscription = MySubscription.joins(:developer).where(:developer_id => dev_id).first
```
Comment (NourhanMohamed) [inline arability/app/models/my_subscription.rb]: lines should not exceed 80 characters, split long lines of code into multiple lines to meet conventions

File: arability/app/models/my_subscription.rb [hunk 32]
```diff
@@ -1,4 +1,84 @@
...
+    old_limit = mySubscription.word_search
+  	if  mySubscription.word_search > 0
+  	  mySubscription.word_search = old_limit - 1
```
Comment (NourhanMohamed) [inline arability/app/models/my_subscription.rb]: leave only one space after the if
rename all variables to match code conventions where all the variable names should consist of lowercase ASCII characters separated by underscores
Example: mySubscription should be my_subscription

File: arability/app/models/my_subscription.rb [hunk 33]
```diff
@@ -1,4 +1,84 @@
...
+#   by returning true else return false and permission denied.
+# params:
+#   developer id
+# success:
+#   permission is given if the developer didnt exceed the search limit
+# fail:
+#   none
+  class << self
+   def search_word_permission(dev_id) 
+    mySubscription = MySubscription.joins(:developer).where(:developer_id => dev_id).first
```
Comment (NourhanMohamed) [inline arability/app/models/my_subscription.rb]: same applies here as far as character limit and variable name conventions

File: arability/app/models/my_subscription.rb [hunk 34]
```diff
@@ -1,4 +1,84 @@
...
+  	if mySubscription.word_search > 0 
+  		return true
+  	else
+  		return false
+    end
+   end
+  end
+# author:Noha
```
Comment (NourhanMohamed) [inline arability/app/models/my_subscription.rb]: add a new line between method end and documentation of the following method

File: arability/app/models/my_subscription.rb [hunk 35]
```diff
@@ -1,4 +1,84 @@
...
+   def add_word_permission(dev_id)
+     mySubscription = MySubscription.joins(:developer).where(:developer_id => dev_id).first
+  	if mysubscription.word_add > 0
+  		return true
+  	else 
+  		return false
+    end  	
+   end
+  end
+# author:Noha
```
Comment (NourhanMohamed) [inline arability/app/models/my_subscription.rb]: same here

File: arability/app/views/projects/_form.html.erb [hunk 36]
```diff
@@ -0,0 +1,21 @@
+<%= form_for(@project) do |f| %>
+  <% if @project.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(@project.errors.count, "error") %> prohibited this project from being saved:</h2>
+
```
Comment (NourhanMohamed) [inline arability/app/views/projects/_form.html.erb:21]: remove extra new line
apply character limit per line

File: arability/app/views/projects/edit.html.erb [hunk 37]
```diff
@@ -0,0 +1,6 @@
+<h1>Editing project</h1>
+
```
Comment (NourhanMohamed) [inline arability/app/views/projects/edit.html.erb]: remove extra new lines

File: arability/app/views/projects/index.html.erb [hunk 38]
```diff
@@ -0,0 +1,21 @@
+<h1>Projects</h1>
+
```
Comment (NourhanMohamed) [inline arability/app/views/projects/index.html.erb]: remove extra new lines

File: arability/config/routes.rb [hunk 39]
```diff
@@ -1,6 +1,19 @@
 Arability::Application.routes.draw do
+
```
Comment (NourhanMohamed) [inline arability/config/routes.rb:8]: too much unnecessary blank lines

**Comment by nohahesham** [comment]
review again
@NourhanMohamed 
@m0hamed 
@Fattouh92 
@SalmaFG

File: arability/app/models/my_subscription.rb [hunk 40]
```diff
@@ -1,4 +1,97 @@
...
+#   takes the developer id and decrements the word search.
+# params:
+#   developer id
+# success:
+#   word search decremented
+# fail:
+#   word search not decremented
+  class << self
+   def decrement_limit(dev_id)
+  	mySubscription =
```
Comment (m0hamed) [inline arability/app/models/my_subscription.rb]: rename variable to my_subscription

File: arability/app/controllers/projects_controller.rb (hunk 28 above)
Comment (NourhanMohamed) [inline arability/app/controllers/projects_controller.rb:1]: remove the 2 comments above the documentation

File: arability/app/models/my_subscription.rb [hunk 41]
```diff
@@ -1,4 +1,97 @@
...
+    MySubscription.joins(:developer).where(:developer_id => dev_id).first
```
Comment (NourhanMohamed) [inline arability/app/models/my_subscription.rb]: when you split lines, you should indent the splitted part by 2 spaces (a tab)
y3ny line 18 should be indented because it belongs el line 17
again the variable names should be changed

File: arability/app/models/my_subscription.rb [hunk 42]
```diff
@@ -1,4 +1,97 @@
...
+# params:
+#   developer id
+# success:
+#   permission is given if the developer didnt exceed the search limit
+# fail:
+#   none
+  class << self
+   def search_word_permission(dev_id) 
+    mySubscription = 
+    MySubscription.joins(:developer).where(:developer_id => dev_id).first
```
Comment (NourhanMohamed) [inline arability/app/models/my_subscription.rb]: same here

File: arability/app/models/my_subscription.rb (hunk 41 above)
Comment (nohahesham) [inline arability/app/models/my_subscription.rb]: i change it to my_subscription?

**Comment by nohahesham** [comment]
@NourhanMohamed

**Comment by NourhanMohamed** [comment]
@nohahesham still there is comments that you didn't change, make sure you change all the comments on the pull request as well as in the issue and then tag me and ashraf to review the code

**Comment by m0hamed** [comment]
also change the labels on the issues otherwise I wouldnt notice

File: arability/app/models/my_subscription.rb [hunk 43]
```diff
@@ -1,4 +1,75 @@
...
+#   by returning true else return false and permission denied.
+# params:
+#   developer id
+# success:
+#   permission is given if the developer didnt exceed the add limit
+# fail:
+#   none
+  class << self
+   def add_word_permission(dev_id)
+     my_Subscription =
```
Comment (m0hamed) [inline arability/app/models/my_subscription.rb]: use small case letters only

File: arability/config/routes.rb [hunk 44]
```diff
@@ -1,6 +1,12 @@
 Arability::Application.routes.draw do
+  resources :projects
+  resources :developer
+  root :to => "Pages#home"
+  get "developer/index"
+
```
Comment (m0hamed) [inline arability/config/routes.rb:4]: avoid unnecessary blank lines

**Comment by NourhanMohamed** [comment]
@nohahesham I don't really know why are u insisting of not editing the pull request and tagging the issue number! 
Please tag this pull request to reference for your issues
we tag issues using # followed by the issue number

File: arability/app/models/my_subscription.rb [hunk 45]
```diff
@@ -1,4 +1,75 @@
...
+#   takes the developer id and checks wether the developer's word search limit has been 
+#   reached ,if its not then it is greater than zero and permission is given to search for words
+#   by returning true else return false and permission denied.
+# params:
+#   developer id
+# success:
+#   permission is given if the developer didnt exceed the search limit
+# fail:
+#   none
+  class << self
```
Comment (NourhanMohamed) [inline arability/app/models/my_subscription.rb]: one tab should be 2 spaces and not one space, please update your indentation spacing accordingly

Comment (NourhanMohamed) [inline arability/app/models/my_subscription.rb]: do this in the rest of the class mesh lazem aktebha f kul method b2a

File: arability/app/models/my_subscription.rb [hunk 46]
```diff
@@ -1,4 +1,75 @@
...
+  		return true
+  	else
+  		return false
+    end
+   end
+  end
+
+# author:Noha hesham
+# Description:
+#   takes the developer id and checks wether the developer's word add limit has been
```
Comment (NourhanMohamed) [inline arability/app/models/my_subscription.rb]: 80 lines MAXIMUM per line, do divide longer lines
This applies for documentation as well as code

File: arability/app/models/my_subscription.rb [hunk 47]
```diff
@@ -1,4 +1,75 @@
...
+# params:
+#   developer id
+# success:
+#   permission is given if the developer didnt exceed the follow word limit
+# fail:
+#   none
+  class << self
+   def follow_word_permission(dev_id)
+    my_Subscription = 
+     MySubscription.joins(:developer).where(:developer_id => dev_id).first
```
Comment (NourhanMohamed) [inline arability/app/models/my_subscription.rb]: same goes for lowercase underscore separated variable names

File: arability/app/models/my_subscription.rb [hunk 48]
```diff
@@ -1,4 +1,75 @@
...
+#   takes the developer id and checks wether the developer's word follow limit has been 
+#   reached ,if its not then it is greater than zero and permission is given to follow words
+#   by returning true else return false and permission denied.
+# params:
+#   developer id
+# success:
+#   permission is given if the developer didnt exceed the follow word limit
+# fail:
+#   none
+  class << self
```
Comment (NourhanMohamed) [inline arability/app/models/my_subscription.rb]: only use one class << self to surround all the methods that use class << self
mesh lazem class << self 7awaleen kul method separately

File: arability/app/views/projects/_form.html.erb [hunk 49]
```diff
@@ -0,0 +1,21 @@
+<%= form_for(@project) do |f| %>
+  <% if @project.errors.any? %>
+    <div id="error_explanation">
+      <h2>
+        <%= pluralize(@project.errors.count, "error") %> prohibited this project from being saved:
```
Comment (NourhanMohamed) [inline arability/app/views/projects/_form.html.erb]: split the line, MAX 80 chars per line

Comment (nohahesham) [inline arability/app/views/projects/_form.html.erb]: sry knt 3amaltaha w neseit a3mel save

Comment (nohahesham) [inline arability/app/views/projects/_form.html.erb]: i did it

**Comment by nohahesham** [comment]
@NourhanMohamed

**Comment by NourhanMohamed** [comment]
comment on the issues not here, and update the labels if u need review
tag ashraf along he's u're reviewer, too
Don't tag either of us to review until all comments are satisfied please
@nohahesham

File: arability/app/controllers/projects_controller.rb [hunk 50]
```diff
@@ -0,0 +1,23 @@
+class ProjectsController < ApplicationController
+  # DELETE /projects/1
+  # DELETE /projects/1.json
```
Comment (m0hamed) [inline arability/app/controllers/projects_controller.rb]: remove the two lines above

File: arability/app/models/my_subscription.rb [hunk 51]
```diff
@@ -1,4 +1,47 @@
...
+# success:
+#  permission is given if the developer didnt exceed the search ,add
+#  or follow limit
+# fail:
+#  none
+
+  class << self
+    def give_permissions(dev_id,type)
+      my_subscription = 
+     MySubscription.joins(:developer).where(:developer_id => dev_id).first
```
Comment (NourhanMohamed) [inline arability/app/models/my_subscription.rb]: MySubscription da eah?

Comment (NourhanMohamed) [inline arability/app/models/my_subscription.rb]: this line should be indented

File: arability/app/views/projects/index.html.erb [hunk 52]
```diff
@@ -0,0 +1,14 @@
+<h1>Projects</h1>
+<table>
+  <tr>
+    <th>Name</th>
+  </tr>
+<% @projects.each do |project| %>
+  <tr>
+    <td><%= project.name %></td>
+    <td><%= link_to 'Destroy', project, method: :delete, data: { confirm: 'Are you sure?' } %></td>
```
Comment (NourhanMohamed) [inline arability/app/views/projects/index.html.erb]: line longer than 80 characters

File: arability/app/models/my_subscription.rb [hunk 53]
```diff
@@ -1,4 +1,47 @@
...
+      if type == 1
+        if my_subscription.word_search > 0
```
Comment (NourhanMohamed) [inline arability/app/models/my_subscription.rb]: greater than zero leeh? ay 7aga greater than zero is permitted?

File: arability/app/models/my_subscription.rb [hunk 54]
```diff
@@ -1,4 +1,47 @@
...
+#  permission is given if the developer didnt exceed the search ,add
+#  or follow limit
+# fail:
+#  none
+
+  class << self
+    def give_permissions(dev_id,type)
+      my_subscription = 
+     MySubscription.joins(:developer).where(:developer_id => dev_id).first
+      if type == 1
```
Comment (NourhanMohamed) [inline arability/app/models/my_subscription.rb]: use constants to denote types

File: arability/app/models/my_subscription.rb [hunk 55]
```diff
@@ -1,4 +1,47 @@
 class MySubscription < ActiveRecord::Base
-  attr_accessible :developer, :word_add, :word_follow, :word_search
+  attr_accessible :developer_id, :word_add, :word_follow, :word_search
   belongs_to :developer
+  belongs_to :subscription_model
+
+  #author:Noha hesham
+# Description:
```
Comment (NourhanMohamed) [inline arability/app/models/my_subscription.rb]: da5aly el documentation gowwa el class << self to be just above the method, with the appropriate indentation

File: arability/app/models/my_subscription.rb (hunk 53 above)
Comment (nohahesham) [inline arability/app/models/my_subscription.rb]: tayeb ma howa hay return true

File: arability/app/models/my_subscription.rb (hunk 54 above)
Comment (nohahesham) [inline arability/app/models/my_subscription.rb]: ya3ni eh mesh fahma?

File: arability/app/models/my_subscription.rb (hunk 51 above)
Comment (nohahesham) [inline arability/app/models/my_subscription.rb]: variable

File: arability/app/controllers/projects_controller.rb [hunk 56]
```diff
@@ -0,0 +1,23 @@
...
+ # Description:
+ #  finds the project by its id then destroys it
+ # params:
+ #  none
+ # success:
+ #  a pop up appears and makes sure the user wants to
+ #  delete the project by choosing ok the 
+ #  project is successfully deleted 
+ # failure:
+ #  * project is not deleted
```
Comment (NourhanMohamed) [inline arability/app/controllers/projects_controller.rb]: remove that asterik and the space that follows

File: arability/app/models/my_subscription.rb [hunk 57]
```diff
@@ -1,4 +1,45 @@
 class MySubscription < ActiveRecord::Base
-  attr_accessible :developer, :word_add, :word_follow, :word_search
+  attr_accessible :developer_id, :word_add, :word_follow, :word_search
   belongs_to :developer
+  belongs_to :subscription_model
+  class << self
+# author:Noha hesham
+# Description:
+#  takes the developer id and integer type and checks wether
```
Comment (NourhanMohamed) [inline arability/app/models/my_subscription.rb]: should be indented by 2 spaces not only one

File: arability/app/models/my_subscription.rb [hunk 58]
```diff
@@ -1,4 +1,45 @@
...
+# params:
+#  developer id and type
+# success:
+#  permission is given if the developer didnt exceed the search ,add
+#  or follow limit
+# fail:
+#  none
+    def give_permissions(dev_id,type)
+      my_subscription = 
+      MySubscription.joins(:developer).where(:developer_id => dev_id).first
```
Comment (NourhanMohamed) [inline arability/app/models/my_subscription.rb]: indent line

File: arability/app/models/my_subscription.rb [hunk 59]
```diff
@@ -1,4 +1,44 @@
...
+    #   and permission is given by returning true else return false
+    #   and permission denied.
+    # params:
+    #   developer id and type
+    # success:
+    #   permission is given if the developer didnt exceed the search ,add
+    #   or follow limit
+    # fail:
+    #   none
+        def give_permissions(dev_id,type)
```
Comment (m0hamed) [inline arability/app/models/my_subscription.rb]: this line should not be indented

also all indents should be two spaces

Comment (NourhanMohamed) [inline arability/app/models/my_subscription.rb]: @nohahesham m3lesh 5ally el def l7ad el end bta3etha tb2a f nafs level fail "raga3eeha tab y3ny"

File: arability/app/views/projects/index.html.erb (hunk 52 above)
Comment (NourhanMohamed) [inline arability/app/views/projects/index.html.erb]: @nohahesham hena line akbar men 80 chars divide it

**Comment by nohahesham** [comment]
@m0hamed 
@NourhanMohamed

**Comment by nohahesham** [comment]
@NourhanMohamed 
@m0hamed  
a pull el master 5alas??

**Comment by nohahesham** [comment]
@NourhanMohamed 
@m0hamed 
mmken 7ad y merge?

**Comment by nohahesham** [comment]
7ad yrod 3alaya?

**Comment by m0hamed** [comment]
closed since this will be reopened for sprint 1
